src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.component.js

```diff
@@ -44,23 +44,54 @@ const useFormDataLifecycle = (
     searchGroupsForSelectedScope,
     addFormIdToReduxStore,
     removeFormDataFromReduxStore,
+    keptFallbackSearchFormValues,
 ) =>
     useEffect(() => {
         // in order for the Form component to render
         // a formId under the `forms` reducer needs to be added.
         searchGroupsForSelectedScope
-            .forEach(({ formId }) => {
-                addFormIdToReduxStore(formId);
+            .forEach(({ formId, searchForm }) => {
+                const elements = searchForm.getElements();
+                const formValuesThatExistInTETypeSearchScope =
+                  ([...elements.values()].reduce((acc, { id: fieldId }) => {
+                      if (Object.keys(keptFallbackSearchFormValues).includes(fieldId)) {
+                          const fieldValue = keptFallbackSearchFormValues[fieldId];
+                          return { ...acc, [fieldId]: fieldValue };
+                      }
+                      return acc;
+                  }, {}));
```

Comment:
This is a little bit complicated. At the moment its is own hook called `useFormDataLifecycle`. 

To explain what we basically doing here I need to explain some bits before. 

So one of those bits is that, when we fallback search this happens on a scope of a program and then broadens up to a scope of a TEType. When this happens basically we push to the `router` a new url.

However we want to keep the search terms that are relevant and have them pre-filed in the form. 

This is what we are doing here. When we instantiate the form we pre-fill it with the relevant data.